dask/array/einsum.py

```diff
@@ -0,0 +1,171 @@
+from numpy.compat import basestring
+
+einsum_symbols = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
+einsum_symbols_set = set(einsum_symbols)
+
+
+def parse_einsum_input(operands):
```

Comment:
please add a note about the source of this function, along the lines of what we see here:
https://github.com/dask/dask/blob/cab9b17174ca61b052f46fd735fa02edb5dc450d/dask/array/numpy_compat.py#L61-L65

Reply:
done